appveyor-linux-binary.yml

```diff
@@ -0,0 +1,295 @@
+version: 1.0.{build}
+image:
+  - Ubuntu2004
+
+configuration:
+  - BuildIntegTesting
+  - LocalZipTerraformBuildIntegTesting
+  - LocalZipTerraformBuildInContainerIntegTesting
+  - S3ZipTerraformBuildIntegTesting
+  - S3ZipTerraformBuildInContainerIntegTesting
+  - OtherTerraformBuildIntegTesting
+  - DeployIntegTesting
+  - PackageIntegTesting
+  - DeleteIntegTesting
+  - SyncIntegTesting
+  - LocalIntegTesting
+  - EndToEndTesting
+  # other Integration testing, Dev, regression and smoke testing
+  - OtherTesting
+
+environment:
+  AWS_DEFAULT_REGION: us-east-1
+  NODE_VERSION: "14.17.6"
+  AWS_S3: 'AWS_S3_TESTING'
+  AWS_ECR: 'AWS_ECR_TESTING'
+  CARGO_LAMBDA_VERSION: "v0.17.1"
+  PYTHON_ARCH: '64'
+  NOSE_PARAMETERIZED_NO_WARN: 1
+  APPVEYOR_CONSOLE_DISABLE_PTY: false
+  APPVEYOR_DETAILED_SHELL_LOGGING: true
+
+  matrix:
+    - PYTHON_HOME: "$HOME/venv3.11/bin"
+      PYTHON_VERSION: '3.11'
+
+install:
+  # AppVeyor's apt-get cache might be outdated, and the package could potentially be 404.
+  - sh: "sudo apt-get update --allow-releaseinfo-change"
+
+  - sh: "gvm use go1.19"
+  - sh: "echo $PATH"
+  - sh: "ls /usr/"
+  # install latest maven which is compatible with jdk17
+  - sh: "sudo apt-get -y remove maven"
+  - sh: "wget https://dlcdn.apache.org/maven/maven-3/3.8.8/binaries/apache-maven-3.8.8-bin.zip -P /tmp"
+  - sh: "sudo unzip -d /opt/mvn /tmp/apache-maven-*.zip"
+  - sh: "PATH=/opt/mvn/apache-maven-3.8.8/bin:$PATH"
+  - sh: "JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64"
+  - sh: "PATH=$JAVA_HOME/bin:$PATH"
+  - sh: "javac -version"
+  - sh: "java -version"
+  - sh: "mvn --version"
+
+  - sh: "source ${HOME}/venv${PYTHON_VERSION}/bin/activate"
+  - sh: "rvm use 2.7.2"
+  - sh: "docker info"
+  - sh: "docker version"
+  - sh: "nvm install ${NODE_VERSION}"
+  - sh: "npm install npm@7.24.2 -g"
+  - sh: "npm -v"
+
+  # Install latest gradle
+  - sh: "sudo apt-get -y remove gradle"
+  - sh: "wget https://services.gradle.org/distributions/gradle-7.3.1-bin.zip -P /tmp"
+  - sh: "sudo unzip -d /opt/gradle /tmp/gradle-*.zip"
+  - sh: "PATH=/opt/gradle/gradle-7.3.1/bin:$PATH"
+
+  # Install AWS CLI
+  - sh: "virtualenv aws_cli"
+  - sh: "./aws_cli/bin/python -m pip install awscli"
+  - sh: "PATH=$(echo $PWD'/aws_cli/bin'):$PATH"
+
+  # Install pytest
+  - sh: "pip3 install -r requirements/pre-dev.txt"
+  - sh: "pip3 install -r requirements/dev.txt"
+  - sh: "pip3 install -r requirements/base.txt"
+  - sh: "pytest --version"
```

Comment:
I think it is better if we installed these libraries in a virtual environment to make sure that we isolate our test cases from any libraries that got installed here. I also think if just install the libraries that are required for the integration testing only.

Reply:
I changed it to download the dependencies into a virtual env, but we require a bunch of these dependencies to run the integration tests. Might not be worth trying to narrow it down right now.